tests/data_out/data_out_14.cc

```diff
@@ -0,0 +1,96 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2003 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+
+#include <deal.II/lac/sparsity_pattern.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include "../tests.h"
+
+#include "data_out_common.h"
+
+namespace
+{
+  template <int dim>
+  struct Wrapper
+  {
+    operator const DataOut<dim> &() const
+    {
+      return data_out;
+    }
+    DataOut<dim> data_out;
+  };
+
+  template <int dim>
+  Wrapper<dim>
+  make_data_out()
+  {
+    DataOut<dim> data_out;
+    return {data_out};
+  }
+
+  template <int dim>
+  Wrapper<dim>
+  make_data_out(const DoFHandler<dim> &dof_handler)
+  {
+    DataOut<dim> data_out = make_data_out<dim>();
```

Comment:
I really had to think about what this all does. In the last line, I *think* this needs to be interpreted is that `make_data_out()` creates an rvalue, you then call the conversion operator, which returns an lvalue reference to `DataOut`, and this then triggers the copy-constructor (but not the move constructor). Is this correct? Perhaps just add a comment in front of that last line here of the form
```
  // Call a function that returns a DataOut object in a way that requires it to be copied into the
  // the data_out object (without the possibility of copy elision or in-place construction):
```